framework/src/systems/NonlinearSystemBase.C

```diff
@@ -2713,6 +2713,12 @@ NonlinearSystemBase::needBoundaryMaterialOnSide(BoundaryID bnd_id, THREAD_ID tid
   return _integrated_bcs.hasActiveBoundaryObjects(bnd_id, tid);
 }
 
+bool
+NonlinearSystemBase::needBoundaryMaterialOnInterface(BoundaryID bnd_id, THREAD_ID tid) const
+{
+  return _integrated_bcs.hasActiveBoundaryObjects(bnd_id, tid);
```

Comment:
This does not get to the root of the problem signified by the BISON segmentation fault (shown below) but I was envisioning this as `_interface_kernels.hasActiveBoundaryObjects`
```
* thread #1, queue = 'com.apple.main-thread', stop reason = EXC_BAD_ACCESS (code=1, address=0x0)
  * frame #0: 0x000000010410e961 libmoose-dbg.0.dylib`MaterialPropertyStorage::initProps(this=0x000000011a8aeef0, material_data=0x000000011a8e2f88, elem=0x000000011a88bf60, side=0, n_qpoints=1, isNeighborOnInterface=true) at MaterialPropertyStorage.C:448
    frame #1: 0x000000010411064d libmoose-dbg.0.dylib`MaterialPropertyStorage::initStatefulProps(this=0x000000011a8aeef0, material_data=0x000000011a8e2f88, mats=size=1, n_qpoints=1, elem=0x000000011a88bf60, side=0, isNeighborOnInterface=true) at MaterialPropertyStorage.C:230
    frame #2: 0x000000010405809f libmoose-dbg.0.dylib`ComputeMaterialsObjectThread::onInterface(this=0x00007fff5fbfe710, elem=0x000000011a88bed0, side=1, bnd_id=16) at ComputeMaterialsObjectThread.C:224
    frame #3: 0x000000010424fb1d libmoose-dbg.0.dylib`ThreadedElementLoopBase<libMesh::StoredRange<libMesh::MeshBase::const_element_iterator, libMesh::Elem const*> >::operator(this=0x00007fff5fbfe710, range=0x00000001171362a0, bypass_threading=true)(libMesh::StoredRange<libMesh::MeshBase::const_element_iterator, libMesh::Elem const*> const&, bool) at ThreadedElementLoopBase.h:225
    frame #4: 0x00000001047e5f53 libmoose-dbg.0.dylib`FEProblemBase::initialSetup(this=0x000000011a003a18) at FEProblemBase.C:587
    frame #5: 0x0000000103dd5657 libmoose-dbg.0.dylib`Transient::init(this=0x000000011a906208) at Transient.C:295
    frame #6: 0x0000000105012e03 libmoose-dbg.0.dylib`MooseApp::executeExecutioner(this=0x0000000118809a18) at MooseApp.C:799
    frame #7: 0x000000010501362b libmoose-dbg.0.dylib`MooseApp::run(this=0x0000000118809a18) at MooseApp.C:904
    frame #8: 0x000000010000221c bison-dbg`main(argc=3, argv=0x00007fff5fbff818) at main.C:49
    frame #9: 0x00007fffcd80c235 libdyld.dylib`start + 1
    frame #10: 0x00007fffcd80c235 libdyld.dylib`start + 1
```

Reply:
Making this change will consequently make the BISON test pass...

Reply:
@lindsayad 
your suggested change makes total sense. However after applying it and running tests on my machine the same error generated in BISON appears in MOOSE for
`interfacekernels/1d_interface.test ............................................................ FAILED (CRASH)`
with BT
```
#0  0x00007ffff588e91f in MaterialPropertyStorage::initProps (this=0x879b40, material_data=..., elem=..., 
    side=0, n_qpoints=1, isNeighborOnInterface=true)
    at /home/arovinelli/projects/moose/framework/src/materials/MaterialPropertyStorage.C:463
#1  0x00007ffff588d3cf in MaterialPropertyStorage::initStatefulProps (this=0x879b40, material_data=..., 
    mats=..., n_qpoints=1, elem=..., side=0, isNeighborOnInterface=true)
    at /home/arovinelli/projects/moose/framework/src/materials/MaterialPropertyStorage.C:230
#2  0x00007ffff5fea720 in ComputeMaterialsObjectThread::onInterface (this=0x7fffffffc120, elem=0x86ad20, 
    side=1, bnd_id=2) at /home/arovinelli/projects/moose/framework/src/loops/ComputeMaterialsObjectThread.C:224
#3  0x00007ffff5364021 in ThreadedElementLoopBase<libMesh::StoredRange<libMesh::MeshBase::const_element_iterator, libMesh::Elem const*> >::operator() (this=0x7fffffffc120, range=..., bypass_threading=true)
    at /home/arovinelli/projects/moose/framework/build/header_symlinks/ThreadedElementLoopBase.h:225
#4  0x00007ffff59a2deb in FEProblemBase::initialSetup (this=0x897610)
    at /home/arovinelli/projects/moose/framework/src/problems/FEProblemBase.C:587
#5  0x00007ffff5cce092 in Steady::init (this=0x92f4c0)
    at /home/arovinelli/projects/moose/framework/src/executioners/Steady.C:50
#6  0x00007ffff6087c56 in MooseApp::executeExecutioner (this=0x6f7080)
    at /home/arovinelli/projects/moose/framework/src/base/MooseApp.C:799
#7  0x00007ffff6087fff in MooseApp::run (this=0x6f7080)
    at /home/arovinelli/projects/moose/framework/src/base/MooseApp.C:904
#8  0x000000000041a808 in main (argc=3, argv=0x7fffffffc7c8)
    at /home/arovinelli/projects/moose/test/src/main.C:31
```
this means that the safeguard I implemented is not working and that `_stateful_prop_id_to_prop_id` in `MaterialPropertyStorage` is not behaving as I was thinking.
There is something else that need to be addressed here. Do you have any clue about the context of `_stateful_prop_id_to_prop_id` (is it defined by element, boundary, block,...)?

Reply:
Have you looked in a debugger at this? Your check is working, but `material_data.props()[0]` is a `nullptr` so calling a method through it results in `EXEC_BAD_ACCESS`. As to the underlying cause, I'm going to investigate a bit